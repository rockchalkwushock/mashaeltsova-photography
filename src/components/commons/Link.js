import React from 'react'
import styled from 'styled-components'
import Link from 'gatsby-link'

// Need to do this for testing purposes.
const UnstyledLink = props => <Link {...props} />

export const ExternalLink = styled.a``

export const InternalLink = styled(UnstyledLink)``

ExternalLink.displayName = 'ExternalLink'
InternalLink.displayName = 'InternalLink'

const SiteLink = ({ children, ext, href, label, text }) =>
  ext ? (
    <ExternalLink href={href} aria-label={label}>
      {children || text}
    </ExternalLink>
  ) : (
    <InternalLink to={href} aria-label={label}>
      {children || text}
    </InternalLink>
  )

export default SiteLink
