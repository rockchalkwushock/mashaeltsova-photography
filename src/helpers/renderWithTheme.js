import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../utils/theme'

/**
 * @fileOverview
 *
 * HOC for testing components with `theme` prop being used.
 */

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
