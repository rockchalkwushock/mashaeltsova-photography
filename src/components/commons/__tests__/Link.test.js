import React from 'react'
import { shallow } from 'enzyme'

import Link, { ExternalLink } from '../Link'
import { renderWithTheme } from '../../../helpers'

describe('Component: <Link />', () => {
  let wrapper
  let wrapper2
  beforeAll(() => {
    wrapper = shallow(
      <Link href="/" label="home page">
        Home
      </Link>
    )
    wrapper2 = shallow(
      <Link ext href="https://cnn.com" label="cnn" text="CNN" />
    )
  })
  test('should render markup as InternalLink', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should render markup as ExternalLink', () => {
    expect(wrapper2).toMatchSnapshot()
  })
  test('should have props as InternalLink', () => {
    expect(wrapper.props()).toHaveProperty('to', '/')
    expect(wrapper.props()).toHaveProperty('aria-label', 'home page')
    expect(wrapper.props()).toHaveProperty('children', 'Home')
  })
  test('should have props as ExternalLink', () => {
    expect(wrapper2.props()).toHaveProperty('href', 'https://cnn.com')
    expect(wrapper2.props()).toHaveProperty('aria-label', 'cnn')
    expect(wrapper2.props()).toHaveProperty('children', 'CNN')
  })
})

describe('Styled Component: <ExternalLink />', () => {
  let wrapper
  beforeAll(() => {
    wrapper = renderWithTheme(
      <ExternalLink href="https://cnn.com" aria-label="hello">
        CNN
      </ExternalLink>
    )
  })
  test('should render markup', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

// REVIEW
// TypeError: Cannot read property 'history' of undefined
// describe('Styled Component: <InternalLink />', () => {
//   let wrapper
//   beforeAll(() => {
//     wrapper = renderWithTheme(
//       <InternalLink to="/" aria-label="hello">
//         Home
//       </InternalLink>
//     )
//   })
//   test('should render markup', () => {
//     expect(wrapper).toMatchSnapshot()
//   })
// })
