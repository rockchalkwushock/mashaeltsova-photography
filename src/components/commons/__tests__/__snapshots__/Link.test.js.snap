// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Link /> should render markup as ExternalLink 1`] = `
<ExternalLink
  aria-label="cnn"
  href="https://cnn.com"
>
  CNN
</ExternalLink>
`;

exports[`Component: <Link /> should render markup as InternalLink 1`] = `
<InternalLink
  aria-label="home page"
  to="/"
>
  Home
</InternalLink>
`;

exports[`Styled Component: <ExternalLink /> should render markup 1`] = `
<ThemeProvider
  theme={
    Object {
      "sizes": Object {
        "lg": "1200px",
        "md": "736px",
        "sm": "540px",
      },
    }
  }
>
  <ExternalLink
    aria-label="hello"
    href="https://cnn.com"
  >
    <a
      aria-label="hello"
      className=""
      href="https://cnn.com"
    >
      CNN
    </a>
  </ExternalLink>
</ThemeProvider>
`;
